### PR TITLE
[DM-35644] Update documentation for new tech notes

### PR DESCRIPTION
**Pull request by rra**
- Add some more from clauses to exceptions
- Revise the documentation for current implementation
- Delete things that were moved

### CODE REVIEW

File: docs/arch/overview.rst [hunk 1]
```diff
@@ -2,6 +2,11 @@
 Overview
 ########
 
+The primary architectural documentation for Gafaelfawr is DMTN-224_, the implementation tech note.
```
Comment (jonathansick) [inline docs/arch/overview.rst:5]: fyi we've got a role that can help with this:

```suggestion
The primary architectural documentation for Gafaelfawr is :dmtn:`224`, the implementation tech note.
```

File: docs/arch/scopes.rst [hunk 2]
```diff
@@ -10,13 +10,17 @@ Each token has zero or more scopes associated with it.
 Protected applications indicate which scopes are required to access that application via the scope parameter on the ``/auth`` route.
 When users create their own tokens, they can choose which scopes to delegate to the user token to restrict its power.
 
-Scopes are derived from the user's group membership as determined by their OpenID Connect claims (if OpenID Connect authentication is used), their GitHub team memberships (if GitHub authentication is used), or their LDAP data (if LDAP is configured).
+Scopes are derived from the user's group membership as determined by their OpenID Connect claims (if OpenID Connect authentication is used), their GitHub team memberships (if GitHub authentication is used), or their LDAP group membership (if LDAP is configured).
 The mapping of groups to scopes is controlled by the ``config.groupMappings`` Helm chart value.
 See :ref:`scopes` for more information.
 
-The list of supported scopes and their human-readable descriptions are configured in the ``known_scopes`` configuration setting.
+The list of supported scopes and their human-readable descriptions are configured in the ``config.knownScopes`` configuration setting.
```
Comment (jonathansick) [inline docs/arch/scopes.rst:17]:
```suggestion
The list of supported scopes and their human-readable descriptions are configured in the ``config.knownScopes`` Helm configuration setting.
```

(Just a note I was initially wondering about the context of this configuration. It's a Helm value, of course, but someone looking at the architectural docs might also be thinking in terms of the in-python configuration.)

File: docs/arch/scopes.rst [hunk 3]
```diff
@@ -26,7 +30,7 @@ Currently, two scopes in that reserved namespace are used:
 * ``admin:token`` grants token administrator powers.
   Users authenticated with a token with this scope can view, create, modify, and delete tokens for any user.
   Administrators are automatically granted this scope when they authenticate.
-  The bootstrap token (configured with the ``bootstrap_token`` configuration setting) is automatically granted ``admin:token`` scope.
+  The bootstrap token (configured with the ``bootstrap-token`` secret) is automatically granted ``admin:token`` scope.
```
Comment (jonathansick) [inline docs/arch/scopes.rst:33]:
```suggestion
  The bootstrap token (configured with the ``bootstrap-token`` Kubernetes secret) is automatically granted ``admin:token`` scope.
```

(sort of the same concern as above)

File: docs/dev/release.rst [hunk 4]
```diff
@@ -47,12 +47,19 @@ In particular, **don't** prefix the tag with ``v``.
 
 The `ci.yaml`_ GitHub Actions workflow uploads the new release to Docker Hub.
 
+3. Create a GitHub release
+--------------------------
+
+Add a new GitHub release for this version.
+The release title should be the same as the version number.
+The description of the release should be the ``CHANGELOG.rst`` entry for this release, converted to Markdown.
```
Comment (jonathansick) [inline docs/dev/release.rst:55]: Can't wait to see you migrate this to Markdown 😆

**Review by jonathansick — APPROVED** [review]

File: docs/dev/release.rst (hunk 4 above)
Comment (rra) [reply docs/dev/release.rst:55]: Maybe I should do that now and get it over with, since I'm in here. :)

**Pull request merged**